### PR TITLE
Framework: Remove deleted packages from CODEOWNERS

### LINKED ISSUE #11812: Framework: CODEOWNERS file and mechanism is broken
@trilinos/framework 

## Bug Report

As reported by @lucbv in #11808, in particular in [this comment](https://github.com/trilinos/Trilinos/pull/11808#issuecomment-1520928275), we used to work with the `CODEOWNERS` file to manage approval rights in PRs. This seems to be broken.

When looking at the `CODEOWNERS` file on GitHub, I can see the following error:

Screen Shot 2023-04-25 at 08 25 53

And in more detail: 

Screen Shot 2023-04-25 at 08 26 02

These errors do apply to all teams, not only the few shown in the screenshots.

⚠️ With the `CODEOWNERS` mechanism being broken, PRs are fully approved by only a single reviewer, not by one reviewer per package/team. 

### Description

One would expect the `CODEOWNERS` file to not contain errors. Furthermore, access rights of teams need to be set correctly, such that the PR approval can be organized by teams again.

### Steps to Reproduce

1. Look at the `CODEOWNERS` file on GitHub to see the above error messages: https://github.com/trilinos/Trilinos/blob/master/.github/CODEOWNERS
1. Look at recent PRs, where a single approval approved the entire PR, although multiple packages have been changed. Example: https://github.com/trilinos/Trilinos/pull/11808

### PR DESCRIPTION
**Pull request by sebrowne**
Also remove commented lines.

@trilinos/framework

## Motivation
Want a valid CODEOWNERS now that we can re-enable CODEOWNERS capability.

## Related Issues

* Closes #11812

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -36,32 +26,20 @@
 
 /packages/kokkos-kernels/    @trilinos/kokkos-kernels
 
-/packages/komplex/           @trilinos/komplex
-
-#/packages/meshinggenie/      @trilinos/
-
-#/packages/minitensor/        @trilinos/
```
Comment (ccober6) [inline .github/CODEOWNERS:43]: minitensor is still package, but is there a difference between having @trilinos/ versus not having anything?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -94,18 +70,10 @@
...
 
-#/packages/ThreadPool/        @trilinos/
-
 /packages/thyra/             @trilinos/thyra
 
 /packages/tpetra/            @trilinos/tpetra
 
-/packages/TriKota/           @trilinos/trikota
-
-#/packages/trilinoscouplings/ @trilinos/
```
Comment (ccober6) [inline .github/CODEOWNERS:105]: Same question for trilinoscouplings.

Comment (sebrowne) [reply .github/CODEOWNERS:105]: Looks like there _is_ a trilinoscouplings team, I'll re-add this and assign them as a writer.

**Review by ccober6 — APPROVED** [review]
lgtm.

**Pull request merged**